src/pb_generator.cc

```diff
@@ -25,38 +25,110 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include "pb_generator.h"
+#include <chrono>
+#include "pb_stub.h"
 
 #include <pybind11/embed.h>
 namespace py = pybind11;
 
 namespace triton { namespace backend { namespace python {
 
 ResponseGenerator::ResponseGenerator(
-    const std::vector<std::shared_ptr<InferResponse>>& responses)
-    : responses_(responses), index_(0)
+    const std::shared_ptr<InferResponse>& response)
+    : id_(response->Id()), is_finished_(false), is_cleared_(false)
 {
+  response_buffer_.push(response);
+}
+
+ResponseGenerator::~ResponseGenerator()
+{
+  if (!is_cleared_) {
+    Clear();
+  }
+  responses_.clear();
 }
 
 std::shared_ptr<InferResponse>
 ResponseGenerator::Next()
 {
-  if (index_ == responses_.size()) {
+  if (is_finished_) {
+    if (!is_cleared_) {
+      Clear();
+    }
     throw py::stop_iteration("Iteration is done for the responses.");
   }
```

Comment:
I was wondering whether we can loop through the responses if we have looped through them once. For example, does the code below work successfully:

```
infer_responses = infer_request.exec(decoupled=True)

for infer_response in infer_responses:
           pass
  
for infer_response in infer_responses:
           pass
```

From the code, it looks like we can only iterate once.

Reply:
Made the iterator to be able to iterate multiple times.